udata/core/dataset/forms.py

```diff
@@ -73,14 +42,8 @@ def __init__(self, label=None, validators=None, **kwargs):
         super(ChecksumField, self).__init__(ChecksumForm, label, validators,
                                             **kwargs)
 
-    def populate_obj(self, obj, name):
-        self._obj = self._obj or Checksum()
-        super(ChecksumField, self).populate_obj(obj, name)
-
-
-class ResourceForm(ModelForm):
-    model_class = Resource
 
+class BaseResourceForm(ModelForm):
```

Comment:
Yup, I'm just surprised I didn't already did this that way :) We discussed this but I probably missed it before merging.